cpp/include/raft/neighbors/cagra.cuh

```diff
@@ -323,16 +323,18 @@ void search(raft::resources const& res,
   auto distances_internal = raft::make_device_matrix_view<float, int64_t, row_major>(
     distances.data_handle(), distances.extent(0), distances.extent(1));
 
-  cagra::detail::search_main<T,
-                             internal_IdxT,
-                             decltype(raft::neighbors::filtering::none_cagra_sample_filter()),
-                             IdxT>(res,
-                                   params,
-                                   idx,
-                                   queries_internal,
-                                   neighbors_internal,
-                                   distances_internal,
-                                   raft::neighbors::filtering::none_cagra_sample_filter());
+  if (1) {  // standard CAGRA
```

Comment:
I've prepared a stub for the runtime dispatch of search variants, please have a look:
https://github.com/rapidsai/raft/pull/2213/files#diff-11ca764d4ee96d5f9fdb500a088130a8772ea8e135274d2493a1912e7d71380bR352-R375